samples/async-messaging-patterns/README.md

```diff
@@ -0,0 +1,289 @@
+     consumer groups to receive a given message. The messages will be load balanced between the consumer groups. This 
+     domain has a blend of message queue and topic behaviour.
+     
+## What You Will Learn  
+   
+- Usage of queues 
+- Usage of topics
+- Usage of durable topics 
+- HTTP to JMS protocol switching 
+
+## Queue Usage Scenario 
+
+### Description
+
+TravelGuideInc intends to provide capability to its users to place travel requests via the web. Currently a Restful 
+service is available to process travel requests. These requests are intended to be dispatched to 
+TravelRequestProcessingService which is managed by a 3rd party travel agent. 
+
+![Queue Usage](queue-usage-scenario.png "Queue Usage")
+
+However, the nature of the 3rd party service (TravelRequestProcessingService) is as follows, 
+
+ - The service becomes unavailable from time to time
+ - The service cannot process more than 300 requests per second
+
+TravelGuideInc has SLA where for its users the following should be provided,
+
+ - 24/7 availability 
+ - delivery guarantees
+ - At a given time more than 500 users could place travel requests. 
+
+By directly proxying the incoming requests to TravelRequestProcessingService would not allow TravelGuideInc to meet 
+its SLAs 
+
+### Solution
+
+As soon as the message is received by the gateway, the message could be placed into a queue. Queue would 
+ensure to persist the message and ensure that the message is successfully delivered to its consumers. 
+
+### Building the Scenario
+
+For this scenario we would need the following,
+
+- Service which will act as the ![TravelRequestProcessingService]("services/samples/queue/"+
+"TravelRequestProcessingService.bal")  
+- Service which will act as the gateway and which will publish the message to the queue 
+![TravelRequestGateway](services/samples/queue/TravelRequestGateway.bal)  
+- Service which will consume the message from the queue and dispatch the message to "TravelRequestProcessingService"
+ ![TravelRequestConsumingService](services/samples/queue/TravelRequestConsumingService.bal)  
+
+Composing the above services queue.balx is created.
+
+### Testing Scenario
+
+#### Sample Setup
+
+1. Start message broker by running the following command,
+
+bin$ ./broker.sh
+
+2. Deploy the service by running the following command,
+
+bin$ ./integrator.sh ../samples/async-messaging-patterns/queue.balx
+
+
+#### Invoking the service
+
+Create a payload.json file with the following content.
+```
+{
+   "Order":{
+      "ID":"797976"
+   },
+   "details":{
+      "Destination":"LHS to UL",
+      "Date":"2017/01/31",
+      "Time":"23:35",
+      "SeatNumber":"14B"
+   }
+}
+```
+Invoke the service using cURL as follows
+
+```
+curl -v http://localhost:9090/travel/itinerary -d@payload.json -H"Content-Type: application/json"
+
+```
+
+#### Observations
+
+The following response will be receive,
```

Comment:
'received' is the past participle

Reply:
Good catch. Fixed 83c6660fc9017b82e21b8a8c1f0e1eee53e8da8b